test-data/unit/check-optional.test

```diff
@@ -428,3 +428,10 @@ x + 1
 [out]
 main:2: note: In module imported here:
 tmp/a.py:3: error: Unsupported left operand type for + (some union)
+
+[case testOptionalFallbackToNone]
+from typing import Any, Optional
+class C(Any):
+    pass
+x = None  # type: Optional[C]
+x.foo()  # E: Some element of union has no attribute "foo"
```

Comment:
Ideas for additional test cases:

* Test case for using `isinstance` with something like `Optional[C]` (`C` has an `Any` base class), and test how the conditional type binder works in that case.
* Similar to above (`isinstance`), but with `Union[Any, int]` where there is no `Any` base class.
* Test accessing an attribute of `Union[Any, int]` where there is no `Any` base class.